### PR TITLE
Upgrade elasticsearch to 8.13.0

### PR DESCRIPTION
**Pull request by KonradMarzec1991**
For `birth_date`, those similarity parameters never worked:
```
// These parameters were previously *always* parsed by TypeParsers#parseField(), even if they
// made no sense; if we've got here, that means that they're not declared on a current mapper,
// and so we emit a deprecation warning rather than failing a previously working mapping.
private static final Set<String> DEPRECATED_PARAMS = Set.of("store", "meta", "index", "doc_values", "index_options", "similarity");
```

For `boost` on `id`, since es8:
```
Index-time boosts have been deprecated since the 5x line, but it was still possible to declare field-specific boosts in the mappings. This is now removed completely. Indexes built in 7x that contain mapping boosts will emit warnings, and the boosts will have no effect in 8.0. New indexes will not permit boosts to be set in their mappings at all.
```

We add boost in query in `_prepare_query_dict`.

### CODE REVIEW

**Comment by wozniakpl** [comment]
Let's not merge it until I don't check the es update on some eph deployment

**Review by patryk-dabrowski — APPROVED** [review]

**Review by pavlo-mk — APPROVED** [review]

**Review by johniak — APPROVED** [review]

**Comment by wozniakpl** [comment]
We'll be merging that in 2.6, so can go on hold now.

File: backend/pyproject.toml [hunk 1]
```diff
@@ -190,7 +191,7 @@ dependencies = [
     "drf-jwt<2.0.0,>=1.19.2",
     "drf-spectacular[sidecar]<1.0,>=0.27",
     "drf-extensions==0.7.1",
-    "elasticsearch-dsl<8.0.0,>=7.0.0",
+    "elasticsearch-dsl==8.13.0",
```
Comment (domdinicola) [inline backend/pyproject.toml:194]: do we want to have <9.0 or <8.14? @KonradMarzec1991

File: backend/pyproject.toml [hunk 2]
```diff
@@ -160,7 +161,7 @@ dependencies = [
     "django-cors-headers<4.0.0,>=3.13.0",
     "django-countries<8.0.0,>=7.3.2",
     "django-csp<4.0,>=3.7",
-    "django-elasticsearch-dsl<8.0.0,>=7.2.2",
+    "django-elasticsearch-dsl==8.0.0",
```
Comment (domdinicola) [inline backend/pyproject.toml:164]: do we want to have <9.0 or <8.1? @KonradMarzec1991

File: backend/pyproject.toml (hunk 1 above)
Comment (KonradMarzec1991) [reply backend/pyproject.toml:194]: Prefer <8.14

File: backend/pyproject.toml (hunk 2 above)
Comment (KonradMarzec1991) [reply backend/pyproject.toml:164]: The same like above, 8.1

**Review by patryk-dabrowski — APPROVED** [review]

**Comment by domdinicola** [comment]
AB#207778

**Comment by domdinicola** [comment]
@wozniakpl can you coordinate to fix conflicts?

**Comment by domdinicola** [comment]
AB#197101

**Comment by domdinicola** [comment]
@patryk-dabrowski to fix conflicts urgent

**Comment by patryk-dabrowski** [comment]
Conflict resolved

**Review by domdinicola — APPROVED** [review]

**Pull request merged**